### PR TITLE
[semver:major] ECS Orb 2.0 Update

### PR DESCRIPTION
**Pull request by KyleTryon**
Changes:
 - MODIFIED:  source structure and development pipeline to mirror Orb Development Kit for easier testing and updating.
   - Shell and Python Scripts have been migrated to the `src/scripts` directory
 - UPDATED:  AWS CLI Orb to 1.3 - *BREAKING CHANGE*
   - Built at [AWS CLI Orb 1.3.1](https://circleci.com/developer/orbs/orb/circleci/aws-cli?version=1.3.1)
   - Now uses AWS CLI v2 by default *BREAKING CHANGE*
   - Expects environment variable names for AWS credentials instead of Strings. ex: `$AWS_SECRET_ACCESS_KEY` -> `AWS_SECRET_ACCESS_KEY`
   - Solves #94 
   - Solves #9 
- MODIFIED: `launch-type` parameter is now an enum, which accepts a blank string as one of the values
- ADDED: `capacity-provider-strategy` parameter
  - Description: The capacity provider strategy to use for the task. If a `capacity-provider-strategy` is specified, the `launch-type` parameter must be set to an empty string.
- ADDED: Usage Example for utilizing Fargate Spot
- ADDED: New Test Job - test-fargatespot
  - Associates the FARGATE and FARGATE SPOT capacity providers with the existing Fargate cluster tests
  - Runs a task with the new custom `capacity-provider-strategy` parameter, enabling the task to run on Fargate SPOT
- MODIFIED: Config organization

### CODE REVIEW

**Comment by KyleTryon** [comment]
I believe I have caused a bug in the github status checks by canceling jobs that were pending on-hold, they are also no longer part of the config. If needed I could potentially temporarily add these and remove them later, but if we can override and merge when ready I think that would be fine.

**Comment by KyleTryon** [comment]
The currently "waiting" checks are not going to complete, we will need to override them. This was caused by pipelines being left on hold and changing the names of workflows since then.

File: src/scripts/verify-revision-is-deployed.sh
Comment (gmemstr) [inline src/scripts/verify-revision-is-deployed.sh:24]: Formatting seems a little weird to me here. Is it cleaner to inline or multiline?

File: src/tests/greet.bats
Comment (gmemstr) [inline src/tests/greet.bats:1]: Invalid test.

**Review by gmemstr — CHANGES_REQUESTED** [review]

File: src/scripts/verify-revision-is-deployed.sh
Comment (KyleTryon) [reply src/scripts/verify-revision-is-deployed.sh:24]: The lines just get a bit long otherwise imo

File: src/examples/update-service.yml
Comment (KyleTryon) [reply src/examples/update-service.yml:8]: instead updated to the real versions.

File: src/README.md
Comment (KyleTryon) [reply src/README.md:1]: As in removing them? I think these are helpful for outside contributors.

Comment (gmemstr) [reply src/README.md:1]: I feel they're overly verbose in a completed orb - it's a bit like commenting

```js
// This function ads things.
const add = (a, b) => a + b; 
```

If we wanted to keep something similar, we could considering combining the content into one file for a single point of reference and to keep the project relatively clean.

Comment (KyleTryon) [reply src/README.md]: removed.

**Review by lokst — COMMENTED** [review]
Great work splitting up the `orb.yml.hbs` file into scripts with `circleci orb pack`! It's much more maintainable now 🎉 I left some comments for your consideration

File: README.MD
Comment (KyleTryon) [reply README.MD:2]: Yes, not sure how those got in comments. removing the comments.

File: src/scripts/run-task.sh
Comment (KyleTryon) [reply src/scripts/run-task.sh:38]: great catch

File: src/scripts/run-task.sh
Comment (KyleTryon) [reply src/scripts/run-task.sh:42]: Indeed I believe so. I am searching for all other instances as well.

File: src/scripts/verify-revision-is-deployed.sh
Comment (KyleTryon) [reply src/scripts/verify-revision-is-deployed.sh:2]: In the event the users passes a string like "$MYVALUE" we would need to eval this to get the true value they intended. That was applied to anything that I felt might be "secretive" or possibly something they may store twice. Because this is just a number, I figured it was unlikely the user would enter something like `$MAX_ATTEMPTS`, though we could add it.

**Comment by lokst** [comment]
@KyleTryon Great work getting to green and thanks for looking into my feedback! Do you think it's feasible to retain the python unit tests?

File: .circleci/config.yml
Comment (KyleTryon) [reply .circleci/config.yml:56]: > @KyleTryon Great work getting to green and thanks for looking into my feedback! Do you think it's feasible to retain running the python unit tests?

Oh yes! sorry! I was distracted by the failed tests.

**Comment by KyleTryon** [comment]
> @KyleTryon Great work getting to green and thanks for looking into my feedback! Do you think it's feasible to retain running the python unit tests?

Tests re-added

**Review by gmemstr — APPROVED** [review]
This looks good to me - it's a pretty beefy PR, so I would love a second pair of eyes on it, but going to go ahead and approve it.

**Review by lokst — APPROVED** [review]
Looks good to me and I see that the tests are passing 👍

**Pull request merged**
